doc/developer/platform/ux.md

```diff
@@ -285,31 +285,111 @@ The HTTPS server exposes two APIs:
 
   * `POST /query` executes a batch of SQL statements.
 
-    The request body specifies the SQL string to execute. The request may contain
-    multiple SQL statements separated by semicolons. The statements are submitted
-    for processing as if they were supplied via the [Simple Query][simple-query]
-    flow of the PostgreSQL protocol (i.e., "the statements are executed as a
-    single transaction unless explicit transaction control commands are included
-    to force a different behavior").
-
-    The response is a JSON array containing the results of each query in the
-    request.
+    The request body is a JSON object containing a key, `sql`, which specifies
+    the SQL string to execute.
+    - The request may contain multiple SQL statements separated by semicolons.
+    - The statements are submitted for processing as if they were supplied via
+    the [Simple Query][simple-query] flow of the PostgreSQL protocol e.g.
+      - the statements are executed as a single transaction unless explicit
+      transaction control commands are included to force a different behavior
+      - execution of the message is abandoned at the first error
+
+    The response is a JSON array containing the in-order results of each
+    executed statement in the request. Notably, this includes the results of
+    statements whose results are ultimately rollback because of an error in a
+    later part of the transaction. Users must parse the results to understand
+    which statements ultimately reflect the resultant state.
 
     In the future, we may expand the feature set of the HTTPS API to allow
     additional control over query execution and result fetching, a la the
     [Snowflake SQL REST API].
 
-    TODO: fully describe the output format.
+    ### Output format
+
+    The output format is a JSON object with one key, `results`, whose values is
+    an array of serialized `adapter::command::SimpleResult` representing all of
+    the executed statements. Some statements might not be executed because we
+    aborted the request early due to encountering an error.
+
+    The serialized `adapter::command::SimpleResult` values are representing in JSON as:
+
+    `SimpleResult` value | JSON value
+    ---------------------|------------
+    `Rows` | `{"rows": <2D array of JSON-ified results>, "col_names": <array of text>}`
+    `Error` | `{"error": <Error string from execution>}`
+    `Ok` | `{ "ok": <tag>}`
+
+    Each committed statement returns exactly one of these values; e.g. in the
+    case of "complex responses", such as `INSERT INTO...RETURNING`, the presence
+    of `Rows` implies `Ok`. In the future we can consider including the tagged
+    `Ok`.
+
+    ### Binding parameters
+
+    You can include parameters alongside the query by including on the request
+    body a key `"parameters"` whose value is an array of objects of the form:
+
+    ```json
+    {
+      "type": "<resolvable type name>",
```

Comment:
It'd be great to allow omitting the `type`, in which case the value gets injected into the query as an untyped string literal—which may or may not work, depending on how much context PostgreSQL can infer.

Reply:
I believe the whole reason it required a type was because we needed to know the type during parsing, and couldn't infer during planning, due to the AST rewrite implementation. I've argued against that, which would alleviate the need for a type here.

Reply:
Yeah, this was just an oversight on my end; the type definition should have always been optional